doc/index.md

```diff
@@ -1,41 +1,30 @@
+-  Transport Layer Security (TLS)- lua-http supports HTTPS and WSS via [luaossl](https://github.com/wahern/luaossl).  
+-  Fully asynchronous API that does not block the current thread when executing operations that typically block
+-  Easy integration into other event-loop based application models
 
+lua-http was written to fill a gap in the Lua ecosystem by providing an HTTP and WebSocket library with the following traits:
+- Asynchronous and performant
+- Can be used without forcing the developer to follow a specific pattern. Conversly, the library can be adapted to many common patterns. 
+- Can be used at a very high level without need to understand the transportation of HTTP data (other than the connection addresses). 
+- Provides a rich low level API, if desired, for creating powerful HTTP based tools at the protocol level.
 
-## Conventions
-
-Operations that may block the current coroutine take an optional timeout.
-
-HTTP 1 request and status line fields are passed around inside of [headers](#http.headers) objects under keys `":authority"`, `":method"`, `":path"`, `":scheme"` and `":status"` as defined in HTTP 2. As such, they are all kept in string form (important to remember for the `:status` field).
-
-Header fields should always be used with lower case keys.
-
-### Errors
-
-Invalid function parameters will throw a lua error (if validated).
-
-Errors are returned as `nil, error, errno` unless noted otherwise.
+As a result of these design goals, the library is simple and un-obtrusive and can accommodate tens of thousands of connections on commodity hardware. 
 
-Some HTTP 2 operations return/throw special [http 2 error objects](#http.h2_error).
+lua-http is a flexible HTTP and WebSocket library that allows developers to concentrate on line-of-business features when building Internet enabled applications. If you are looking for a way to streamline development of an internet enabled application, enable HTTP networking in your game, create a new Internet Of Things (IoT) system, or write a performant custom web server for a specific use case, lua-http has the tools you need.
 
-
-## Terminology
-
-Much lua-http terminology is borrowed from HTTP 2.
-
-A ["connection"](#connection) is an abstraction over the underlying socket.
-lua-http has two connection types: one for HTTP 1, one for HTTP 2.
-
-A ["stream"](#stream) is a request/response on a connection.
-lua-http has two stream types: one for [HTTP 1 streams](#http.h1_stream), and one for [HTTP 2 streams](#http.h2_stream).
-They share a lowest common denominator interface, see [*stream*](#stream) and [*http.stream_common*](#http.stream_common).
+[1] _lua-http is pure lua code and will therefore support any platform that Lua 5.1 or greater supports. Where lua-http can run is mainly limited by where cqueues works (which is BSDs, Linux, OSX, Solaris): if you can port cqueues to it, lua-http should automatically work._
```

Comment:
avoid statements of fact that may change in future.
"which is BSDs" => "which at the time of writing is BSDs"